### DISCUSSION TITLE
i3blocks not recognizing commands

### DISCUSSION THREAD

**Discussion by guth0** [i3/i3#6130] [Q&A]
I am configuring my i3blocks for my bottom bar and ran into some problems while trying to display my current keyboard layout. I decided to use the command "setxkbmap -query" to get the layout, but every time I try to run i3blocks, I get an error: 

"Command "setxkbmap -query | grep layout | awk -F ' ' '{print $2}' " not found or missing dependency.

I have already tested this by running it in a terminal so I know that I do actually have "setxkbmap" installed and working as well as grep and awk. 

I have also ran i3blocks with just "setxkbmap -query" as the command and it gives the same result so I know that is where the problem is coming from.

Other programs like "localectl status" also give the same error. 

I also have other modules that do work that use programs like "pactl" and "ip" so I know that it does not happen to everything.

Anything helps, thank you :)

edit: I don't know why, but running a bash script through C++ worked. Although I am still having problems. While other parts of my bar update when I send a signal to "-SIGRTMIN+10", this module does not. And yes, it does have all the same interval and signal settings.

**Comment by okraits** [comment]
Can you please provide the configuration of the block which doesn't work?

**Reply by guth0** [reply]
> Can you please provide the configuration of the block which doesn't work?

Block:
```
[kb-layout]
command="'$HOME/.config/i3/getKB/getKB.o"
interval=5
signal=10

```
getKB.cpp:
```
#include <iostream>

int main() {
    system("$HOME/.config/i3/getKB/getKB.sh");
    return 0;
}
```

getKB.sh:
```
#!/bin/env bash

keymap="$(setxkbmap -query | grep layout | awk -F ' ' '{print $2}')"
echo "$keymap"
exit 0
```

I have no idea why, but this is the only way I could get it to work. Running just `getKB.sh` or just the commands inside of it did not work. 

Although it is super janky, I don't REALLY need a fix for that. Now, I mainly just want to figure out why the signal does not work.

Here is another one of my blocks that successfully uses the signal
```
[volume]
command=echo "VOL: $(pactl get-sink-volume @DEFAULT_SINK@ | awk -F "/" '{print $2}' | sed -n 's/^\s*\([0-9]\+\)%/\1%/p' | sed 's/ $//')"
interval=5
signal=10
```

**Comment by okraits** [comment]
Why is there a single quote in the command of the first block you showed? Please also make sure that the binary you are using is executable.
